seastar_cmake.py

```diff
@@ -21,10 +21,15 @@
 
 ROOT_PATH = os.path.realpath(os.path.dirname(__file__))
 
-BUILD_PATHS = { mode: os.path.join(ROOT_PATH, 'build', mode) for mode in SUPPORTED_MODES }
-
 COOKING_BASIC_ARGS = ['./cooking.sh']
 
+def build_path(mode, build_root = 'build'):
+    """Return the absolute path to the build directory for the given mode,
+    i.e., seastar_dir/<build_root>/<mode>"""
+    if mode not in SUPPORTED_MODES:
+        raise RuntimeError('Mode ' + mode + ' is not a supported mode')
```

Comment:
`mode` is always `in SUPPORTED_MODES`, as long as user uses `seastar_cmake.py` via `configure.py` or `test.py`. as both these two command line tools enforces this using something like
```py
parser.add_argument('--mode', choices=seastar_cmake.SUPPORTED_MODES, help="Run only tests for given build mode")
```

and expand the `mode` into all supported mode if it is `all`. so, what we might want to do here instead is to verify this by
```py
assert mode in SUPPORTED_MODES, f'Mode {mode} is not supported')
```

as otherwise, it'd be an unexpeceted programming error.

Reply:
Good call, I had missed that `choices` restricted the input to the valid set. I've changed it to be an assert.